### ISSUE TITLE
River fails to compile when wlroots is built without x11 backend

### ISSUE THREAD

**Issue by epsilon-0** [ifreund/river#231]
Hi,
 It seems tha wlroots API change means that River is no longer able to be built, I get the following errors on compile:
```
ld.lld: error: undefined symbol: wlr_xdg_surface_for_each_popup
>>> referenced by xdg_shell.zig:325 (/var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-wlroots/src/types/xdg_shell.zig:325)
>>>               /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/4bb29bf04f0592e88bffbf44ce1c0834/river.o:(.wayland.wayland_server_core.struct:490:17.wrapper.83)
>>> referenced by xdg_shell.zig:325 (/var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-wlroots/src/types/xdg_shell.zig:325)
>>>               /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/4bb29bf04f0592e88bffbf44ce1c0834/river.o:(.wayland.wayland_server_core.struct:490:17.wrapper.83)

ld.lld: error: undefined symbol: wlr_layer_surface_v1_for_each_popup
>>> referenced by layer_shell_v1.zig:123 (/var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-wlroots/src/types/layer_shell_v1.zig:123)
>>>               /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/4bb29bf04f0592e88bffbf44ce1c0834/river.o:(.wayland.wayland_server_core.struct:490:17.wrapper.83)
>>> referenced by layer_shell_v1.zig:123 (/var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-wlroots/src/types/layer_shell_v1.zig:123)
>>>               /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/4bb29bf04f0592e88bffbf44ce1c0834/river.o:(.wayland.wayland_server_core.struct:490:17.wrapper.83)
>>> referenced by layer_shell_v1.zig:123 (/var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-wlroots/src/types/layer_shell_v1.zig:123)
>>>               /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/4bb29bf04f0592e88bffbf44ce1c0834/river.o:(.wayland.wayland_server_core.struct:490:17.wrapper.83)
>>> referenced 1 more times

ld.lld: error: undefined symbol: wlr_output_is_x11
>>> referenced by Output.zig:563
>>>               /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/4bb29bf04f0592e88bffbf44ce1c0834/river.o:(.wayland.wayland_server_core.struct:490:17.wrapper.85)

ld.lld: error: undefined symbol: wlr_x11_output_set_title
>>> referenced by Output.zig:564
>>>               /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/4bb29bf04f0592e88bffbf44ce1c0834/river.o:(.wayland.wayland_server_core.struct:490:17.wrapper.85)
error: LLDReportedFailure
river...The following command exited with error code 1:
/usr/bin/zig build-exe /var/tmp/portage/gui-wm/river-9999/work/river-9999/river/main.zig --library c --library evdev --library wayland-server --library xkbcommon --library pixman-1 --library wlroots -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/xdg-shell-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/pointer-gestures-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/xdg-output-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/pointer-constraints-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/river-control-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/river-options-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/river-status-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/wlr-layer-shell-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/wlr-output-power-management-unstable-v1-protocol.c --pkg-begin build_options /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/river_build_options.zig --pkg-end -OReleaseSafe --cache-dir /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache --global-cache-dir /var/tmp/portage/gui-wm/river-9999/homedir/.cache/zig --name river --pkg-begin wayland /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/wayland.zig --pkg-end --pkg-begin xkbcommon /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-xkbcommon/src/xkbcommon.zig --pkg-end --pkg-begin pixman /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-pixman/pixman.zig --pkg-end --pkg-begin wlroots /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-wlroots/src/wlroots.zig --pkg-begin wayland /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/wayland.zig --pkg-end --pkg-begin xkbcommon /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-xkbcommon/src/xkbcommon.zig --pkg-end --pkg-begin pixman /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-pixman/pixman.zig --pkg-end --pkg-end -I /usr/include/libevdev-1.0 -I /usr/lib64/libffi/include -I /usr/include/pixman-1 -I /usr/lib64/libffi/include -I /usr/include/libdrm -I /usr/include/pixman-1 -I /usr/include/elogind -I /usr/include/uuid --enable-cache
error: the following build command failed with exit code 1:
/var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/2c70ea06238f3b5a146d42c618fd3f95/build /usr/bin/zig /var/tmp/portage/gui-wm/river-9999/work/river-9999 /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache /var/tmp/portage/gui-wm/river-9999/homedir/.cache/zig -Drelease-safe
```

Let me know if I can provide more information to help with this <3

Aisha

**Comment by Leon-Plickat** [comment]
That is expected. River is developed against wlroots releases, not git master.

**Comment by epsilon-0** [comment]
I tried building with latest wlroots release 0.12.0 and it also gave a compilation error
```
>>> Compiling source in /var/tmp/portage/gui-wm/river-9999/work/river-9999 ...
ld.lld: error: undefined symbol: wlr_output_is_x11
>>> referenced by Output.zig:563
>>>               /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/4bb29bf04f0592e88bffbf44ce1c0834/river.o:(.wayland.wayland_server_core.struct:490:17.wrapper.85)

ld.lld: error: undefined symbol: wlr_x11_output_set_title
>>> referenced by Output.zig:564
>>>               /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/4bb29bf04f0592e88bffbf44ce1c0834/river.o:(.wayland.wayland_server_core.struct:490:17.wrapper.85)
error: LLDReportedFailure
river...The following command exited with error code 1:
/usr/bin/zig build-exe /var/tmp/portage/gui-wm/river-9999/work/river-9999/river/main.zig --library c --library evdev --library wayland-server --library xkbcommon --library pixman-1 --library wlroots -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/xdg-shell-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/pointer-gestures-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/xdg-output-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/pointer-constraints-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/river-control-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/river-options-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/river-status-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/wlr-layer-shell-unstable-v1-protocol.c -cflags -std=c99 -- /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/wlr-output-power-management-unstable-v1-protocol.c --pkg-begin build_options /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/river_build_options.zig --pkg-end -OReleaseSafe --cache-dir /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache --global-cache-dir /var/tmp/portage/gui-wm/river-9999/homedir/.cache/zig --name river --pkg-begin wayland /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/wayland.zig --pkg-end --pkg-begin xkbcommon /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-xkbcommon/src/xkbcommon.zig --pkg-end --pkg-begin pixman /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-pixman/pixman.zig --pkg-end --pkg-begin wlroots /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-wlroots/src/wlroots.zig --pkg-begin wayland /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/zig-wayland/wayland.zig --pkg-end --pkg-begin xkbcommon /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-xkbcommon/src/xkbcommon.zig --pkg-end --pkg-begin pixman /var/tmp/portage/gui-wm/river-9999/work/river-9999/deps/zig-pixman/pixman.zig --pkg-end --pkg-end -I /usr/include/libevdev-1.0 -I /usr/lib64/libffi/include -I /usr/include/pixman-1 -I /usr/lib64/libffi/include -I /usr/include/libdrm -I /usr/include/pixman-1 -I /usr/include/elogind --enable-cache
error: the following build command failed with exit code 1:
/var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache/o/2c70ea06238f3b5a146d42c618fd3f95/build /usr/bin/zig /var/tmp/portage/gui-wm/river-9999/work/river-9999 /var/tmp/portage/gui-wm/river-9999/work/river-9999/zig-cache /var/tmp/portage/gui-wm/river-9999/homedir/.cache/zig -Drelease-safe
```
Does this need an even older release?

**Comment by ifreund** [comment]
No you shouldn't need an older release, 0.12.0 is correct. It looks like your wlroots is built without the x11 backend enabled though. 

Unfortunately there isn't currently a way to avoid this requirement. What river should do is check if wlroots was built with support for the x11 backend at build time and change behavior accordingly.

**Comment by ifreund** [comment]
Should be fixed by 27cac16ced8ec4a93ac6dec224a4ccabd70f4659
